label_studio/ml_models/models.py

```diff
@@ -173,9 +173,16 @@ def error_file_name(self):
     def delete_predictions(self):
         """
         Deletes any predictions that have originated from a ModelRun
-        """
 
-        Prediction.objects.filter(model_run=self.id).delete()
+        Executing a raw SQL query here for speed. This ignores any foreign key relationships
+        so if another model has a Prediction fk and set to on_delete=CASCADE for example,
+        it will not take affect. The only relationship like this that currently exists
+        is in Annotation.parent_prediction, which we are handling here
+        """
+        predictions = Prediction.objects.filter(model_run=self.id)
+        prediction_ids = [p.id for p in predictions]
+        Annotation.objects.filter(parent_prediction__in=prediction_ids).update(parent_prediction=None)
```

Comment:
I think we can do an on_delete update to None/null in django?
parent = models.ForeignKey(Parent, on_delete=models.SET_NULL, null=True)

Reply:
oh we are moving away from delete to speedup